### ISSUE TITLE
nuget update still fails for new csproj format

### ISSUE THREAD

**Issue by nietras** [NuGet/Home#6382]
When will this be resolved? What if one needs to use the `PackageReference` and cannot use `packages.config`?

## Details about Problem
Running:
```
nuget.exe update NewCsproj.csproj
```
fails with
```
Unable to update. The project does not contain a packages.config file.
```

NuGet Version: 4.4.1.4656
VS version (if appropriate): 15.5.2
MSBuild version: 15.5.180.51428 for .NET Framework
OS version (i.e. win10 v1607 (14393.321)): win10

## Detailed repro steps so we can see the same problem

1. Run `nuget.exe update` on any new csproj file with `PackageReference` e.g.:
```
  <ItemGroup>
    <PackageReference Include="SomeNugetPackage" Version="1..0.0-alpha" />
  </ItemGroup>
```

**Comment by zhili1208** [comment]
Could you try donet Cli: https://github.com/dotnet/cli ? "donet add [csproj] package [packageName]", it should update the package for PackageReference.

**Comment by Indomitable** [comment]
Hi, I have two classic .csproj projects that use PackageReference both do not have packages.config file on one of them nuget update works, on the other it fails with "Unable to update. The project does not contain a packages.config file." 
When Nuget searches for packages.config file ?

**Comment by AngelosP** [comment]
@zhili1208 any updates on this? Here is the current behavior:

The command doesn't fail or give any warnings:
![image](https://user-images.githubusercontent.com/8742622/35350305-5db7368c-00f2-11e8-9425-0f11ad6f6e98.png)

But if you go back into Visual Studio it's clear that certain projects were completely skipped and are still referencing the old version of the component:
![image](https://user-images.githubusercontent.com/8742622/35350152-daf989e8-00f1-11e8-9690-f78d1943fde3.png)

**Comment by jainaashish** [comment]
NuGet.exe commands like install or update aren't supported for `PackageReference` based projects. They are only supported for `packages.config` based projects. Instead please use `dotnet add` to add or update packages for `PackageReference` for now. And we also have a open workitem https://github.com/NuGet/Home/issues/4103 to explicitly support `dotnet update` command. So you can upvote on that existing issue.

**Comment by nietras** [comment]
```
dotnet add NewCsproj.csproj package SomeNugetPackage
```
does not update. If say 1.2.4 is installed, and a 1.2.5 is available then this command will not update to 1.2.5. If however I write:
```
dotnet add NewCsproj.csproj package SomeNugetPackage -v 1.2.5
```
It will change to 1.2.5. But I can't do that in my scenario, I need to update to latest version.

**Comment by nietras** [comment]
What you can do then is to remove and then update:
```
dotnet remove NewCsproj.csproj package SomeNugetPackage
dotnet add NewCsproj.csproj package SomeNugetPackage
```
